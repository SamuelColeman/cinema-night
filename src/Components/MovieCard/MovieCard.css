.movie_card {
	border: 1px solid black;
	width: 31%;
	text-align: center;
	background-color: #E1E1E1;
	margin-top: 1%;
}

h1 {
	text-align: center;
  font-size: 30px;
}

.movie__card--img {
	width: 79%;
}

button {
	display: block;
}