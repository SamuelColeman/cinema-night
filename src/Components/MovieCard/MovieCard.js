import React from 'react';

const MovieCard = ({ title, poster_path }) => {

    return (
        <section>
            <p>{ title }</p>
            <img src={poster_path} />
        </section>
    )
} 

export default MovieCard;