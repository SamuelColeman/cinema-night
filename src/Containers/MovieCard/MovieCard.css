a {
	text-decoration: none;
}

.movie_card {
	border: 1px solid white;
	width: 31%;
	text-align: center;
	margin-top: 4%;
	transition: transform .2s;
	background: rgba(48,52,63,.7)
}

.isFavorited {
	background-color:red;
	border: 3px solid red
}

.movie_card:hover {
	transform: scale(1.1); 
	}

.movie__card--title {
	text-align: center;
	font-size: 1.75em;
	margin-top: 0;
	color: #CFCBCA;
	text-shadow: 4px 15px #30343f;
}

.movie__card--img {
	width: 79%;
}

button:hover {
	background: white;
	color: black;
}

.active {
	background: red;
	max-width: 19%
}

section > img {
	background-color: rgba(207,203,202,.7);
	width: 12%;
	display: inline-block;
}

section > img:hover {
	cursor: pointer
}

.movie__h1--error {
	color: red;
	font-size: 1em;
	margin-top: 2%;
	margin-right: 9%;
	font-style: italic;
}

.movie__div--fave {
	display: flex;
	justify-content: flex-end;
}