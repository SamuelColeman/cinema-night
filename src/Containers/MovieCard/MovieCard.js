import React from 'react';
import './MovieCard.css';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import fave from '../../images/transparent-heard.png'

export const MovieCard = ({ title, poster_path, overview, id, movies, hasError, handleFavourite, active }) => {
    let currentMovie =  movies.find(movie => movie.id === id);
    let postedMovie = { 
        movie_id: currentMovie.id,
        title: currentMovie.title,
        poster_path: currentMovie.poster_path,
        release_date: currentMovie.release_date,
        vote_average: currentMovie.vote_average,
        overview: currentMovie.overview
     }

        return (
           <section className="movie_card" id={id}> 
                <img src={fave} id={currentMovie.title} className= {active} onClick={() => handleFavourite(postedMovie)}></img>

                <Link to={`/movies/${id}`}>
                    <div>
                        <h1>{ title }</h1>
                        <img className="movie__card--img" src={poster_path} alt={title} />
                        <h1>{ hasError }</h1>
                    </div>
                </Link>    
            </section>
        )
}   

export const mapStateToProps = (state) => ({
    movies: state.movies,
    hasError: state.hasError
})

export default connect(mapStateToProps)(MovieCard);