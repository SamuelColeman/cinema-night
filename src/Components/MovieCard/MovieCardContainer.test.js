import React from 'react';
import { shallow } from 'enzyme';
import { MovieCard, mapStateToProps, mapDispatchToProps } from './MovieCard';

describe('MovieCardContainer', () => {

    describe('MovieCard', () => {
        let wrapper;
        let mockFunction = jest.fn();
        let chosenMovie =  {
            id: 456,
            title: 'Harry Potter and the Sorceres Stone',
            popularity: 462.91,
            vote_count: 3775, 
            video: false,
          }
    
        beforeEach(() => {
          wrapper = shallow(<MovieCard 
            key={chosenMovie.id}
            selectMovie={mockFunction}
            {...chosenMovie}
          />)
        })
    
        it('should match snapshot with all data passing through correctly', () => {
            expect(wrapper).toMatchSnapshot();
        })
     })

     describe('mapStateToProps', () => {

     })
})