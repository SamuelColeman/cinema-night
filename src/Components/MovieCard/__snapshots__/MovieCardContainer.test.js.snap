// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardContainer MovieCard should match snapshot with all data passing through correctly 1`] = `undefined`;
